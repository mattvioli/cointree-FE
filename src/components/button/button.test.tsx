import { render, screen, fireEvent } from "@testing-library/react";
import { Button } from "./index";

describe("Button component", () => {
  it("renders the button text correctly", () => {
    const buttonText = "Click Me";
    render(<Button onClick={() => {}}>{buttonText}</Button>);

    const buttonElement = screen.getByRole("button");

    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toHaveTextContent(buttonText);
  });

  it("calls the onClick handler when the button is clicked", () => {
    const onClickMock = jest.fn();
    const buttonText = "Click Me";
    render(<Button onClick={onClickMock}>{buttonText}</Button>);

    const buttonElement = screen.getByRole("button");

    fireEvent.click(buttonElement);

    expect(onClickMock).toHaveBeenCalledTimes(1);
  });

  it("has the correct Tailwind classes applied", () => {
    const buttonText = "Click Me";
    render(<Button onClick={() => {}}>{buttonText}</Button>);

    const buttonElement = screen.getByRole("button");

    expect(buttonElement).toMatchSnapshot();
  });
});
