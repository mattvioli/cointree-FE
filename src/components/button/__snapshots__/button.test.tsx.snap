// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component has the correct Tailwind classes applied 1`] = `
<button
  class="rounded-full px-6 bg-black text-white tracking-tight cursor-pointer h-12"
>
  Click Me
</button>
`;
